File: ClientManagerServer/ClientManagerServer/Models/UserModel.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ClientManagerServer.Models
{
    public class UserModel
    {
        public string id { get; set; }
        public string firstName { get; set; }
        public string lastName { get; set; }
        public string email { get; set; }
        public string address { get; set; }
    }
}

File: ClientManagerClient/src/app/model/userModel.ts
import * as actions from '../actions/userActions';

export class UserModel {
    id: string;
    firstName: string;
    lastName: string;
    email: string;
    address: string;
}


export function userReducer(state: Array<UserModel> = [], act: { type: string, payLoad: any }): Array<UserModel> {
    switch (act.type) {
        case (actions.GET_USERS_SUCC):
            return act.payLoad;
        default:
            return state;
    }
}

export function selectedUserReducer(state: UserModel, act: { type: string, payLoad: any }): UserModel {
    switch (act.type) {
        case (actions.GET_USER_SUCC):
            return act.payLoad;
        default:
            return state;
    }
}

